internal/impl/confluent/serde_goavro.go

```diff
@@ -121,8 +123,22 @@ func (s *schemaRegistryEncoder) getAvroEncoder(ctx context.Context, schema franz
 	}, nil
 }
 
-func (s *schemaRegistryDecoder) getGoAvroDecoder(ctx context.Context, schema franz_sr.Schema) (schemaDecoder, error) {
-	schemaSpec, err := resolveGoAvroReferences(ctx, s.client, s.cfg.avro.mapping, schema)
+func tryExtractCommonSchemaFromAvroBytes(rawUnion bool, specBytes []byte) (any, error) {
+	var as any
+	if err := json.Unmarshal(specBytes, &as); err != nil {
+		return nil, err
+	}
+
+	commonSchema, err := avroAnyToCommonSchema(rawUnion, as)
```

Comment:
Could we avoid passing the bool parameter from function to function? Maybe adding a type would help.